### PR TITLE
Add Mock HTTP Tests for Akamai Validator 

### PR DESCRIPTION
**Pull request by fbeaty4**
## Changes
Added mock http tests for akamai validator
Also reworked the validator request authentication header piece based on Akamai code found here: 
https://github.com/akamai/AkamaiOPEN-edgegrid-C-Sharp

For significant contributions please make sure you have completed the following items:

* [ ] `ReleaseHistory.md` updated for non-trivial changes
* [x ] Added unit tests

### CODE REVIEW

File: Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs [hunk 1]
```diff
@@ -14,15 +15,84 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
...
+            string requestData = GetRequestData(request.Method.ToString(), request.RequestUri);
+            string authData = GetAuthDataValue(id, resource, timestamp, scanIdentityGuid);
+            string authHeader = GetAuthorizationHeaderValue(secret, timestamp, authData, requestData);
+            request.Headers.Add("Authorization", authHeader);
+
+            return request;
+        }
+
+        internal static string GetRequestData(string method,
+                                              Uri uri)
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs:36]: not all methods should follow this style.
only the ones that gets too long.

in this case, you can keep in one line.

File: Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs [hunk 2]
```diff
@@ -14,15 +15,84 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
...
+        {
+            using (var algorithm = HMAC.Create(hashType.ToString()))
             {
-                return ValidationResult.CreateNoMatch();
+                algorithm.Key = Encoding.UTF8.GetBytes(key);
+                return algorithm.ComputeHash(data);
             }
+        }
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs:90]: this is a good max length. if any method is longer than this, we should break in multiple lines

File: Src/Plugins/Tests.Security/Validators/SEC101_015.AkamaiCredentialsValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+            foreach (HttpMockTestCase testCase in testCases)
+            {
+                for (int i = 0; i < testCase.HttpRequestMessages.Count; i++)
+                {
+                    mockHandler.Mock(testCase.HttpRequestMessages[i], testCase.HttpResponseMessages[i]);
+                }
+
+
+
```
Comment (eddynaka) [inline Src/Plugins/Tests.Security/Validators/SEC101_015.AkamaiCredentialsValidatorTests.cs:92]: same here

File: Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs [hunk 4]
```diff
@@ -14,15 +15,84 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
...
-        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
+        internal static HttpRequestMessage GenerateRequestMessage(string id,
+                                                                  string host,
+                                                                  string secret,
+                                                                  string resource,
+                                                                  string scanIdentityGuid,
+                                                                  DateTime datetime)
+        {
+            string timestamp = datetime.ToString("yyyyMMdd'T'HH:mm:ss+0000");
+            using var request = new HttpRequestMessage(HttpMethod.Get, $"{host}/ccu/v2/queues/default");
```
[the remark this answers is not in the document]
Comment (fbeaty4) [reply Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs]: removed

File: Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs [hunk 5]
```diff
@@ -14,15 +15,84 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
...
+
+            byte[] data = Encoding.UTF8.GetBytes(string.Format("{0}{1}", requestData, authData));
+            string authSignature = Convert.ToBase64String(ComputeKeyedHash(data, signingKey, hashType));
+
+            return string.Format("{0}signature={1}", authData, authSignature);
+        }
+
+        internal static byte[] ComputeKeyedHash(byte[] data,
+                                                string key,
+                                                string hashType)
```
[the remark this answers is not in the document]
Comment (fbeaty4) [reply Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs]: got it!

**Pull request merged**